### ISSUE TITLE
IDISP014 should not warn here

### ISSUE THREAD

**Issue by dominicshaw** [DotNetAnalyzers/IDisposableAnalyzers#133]
In this simple class you get IDISP014
```cs
    public class BrokerCommissionsDownloader
    {
        private static readonly HttpClient _client;

        static BrokerCommissionsDownloader()
        {
            _client = new HttpClient(new HttpClientHandler() {UseDefaultCredentials = false}) {BaseAddress = new Uri("http://server/")};
        }
```
this is semantically the same as
```cs
    public class BrokerCommissionsDownloader
    {
        private static readonly HttpClient _client = new HttpClient(new HttpClientHandler() {UseDefaultCredentials = false}) {BaseAddress = new Uri("http://server/")};
```
which does not produce the same warning. Note that I do have another HttpClient in my project elsewhere but I don't think that is why I am getting the warning - I believe it is the static constructor that confuses the analyser.

**Comment by JohanLarsson** [comment]
You are right, I was lazy analyzing this only allowing field & property initializers. It is reasonable to teach the analyzer static ctors. Thanks for reporting!
